packages/compiler/src/core/checker.ts

```diff
@@ -102,6 +107,14 @@ import {
   StdTypes,
   StringLiteral,
   StringLiteralNode,
+  StringTemplate,
+  StringTemplateExpressionNode,
+  StringTemplateHeadNode,
```

Comment:
Do you split this into head/middle/end so that you only capture the part that needs to be computed as the middle?

Reply:
Those are technically slightly different nodes and that's how typescript also structure their template so thought be good to keep inline as they must have had a reason to do that. But right now the data inside is basically the same